source/blueprints/banking.rst

```diff
@@ -0,0 +1,546 @@
+        else:
+            if(_get('auto_pay') != 0):
+                replies = ['AutoPay is already turned on. To turn off just say \'autopay off\'.']
+            else:
+                replies = ['AutoPay has been turned on. To turn off just say \'autopay off\'.']
+                _put('auto_pay', 1)
+        # respond with the current state of autopay
+        responder.reply(replies)
+
+
+This code is a simpler example of how to incorporate a third party API, the ``_get`` and ``_put`` methods are used here to mimic an API calls to get information and push an update. 
+
+Here we will illustrate a simple version of the ``pay_creditcard`` handler, which is more complex than the prior handler and uses custom and system entities:
+
+.. code:: python
+
+    @app.handle(intent='pay_creditcard')
+    def setup_autopay(request, responder):
+        """
+        When the user expresses an intent to make a credit card payment, check to see if
+	they have specified a numerical amount or a generic amount like minimum
+        """
+	#check to see what entity the user has specified 
+        for entity in request.entities:
+	    #if it is a credit amount type entity we can narrow it down to being balance or minimum
+            if entity['type'] == 'credit_amount':
+                responder.slots['payment'] = entity['value'][0]['cname'] or entity['text']
+		#if it is minimum proceed to handle a payment for the minimum amount of the total credit balance 
+                if(responder.slots['payment'] == 'minimum'):
+                    responder.reply(['Ok we have scheduled your credit card payment for your {payment} balance of ${min}'])
+		    #update the user data with the new amount for the credit balance 
+                    _put('credit', _get('credit') - responder.slots['min'])
+		    #update the checking account to reflect the payment made 
+		    _put('checking', get('checking') - responder.slots['min'])
+		#if it is balance handle it similar to minimum but with the amount being the total credit debt
+                else:
+                    responder.reply(['Sure, we have scheduled your credit card payment for your {payment} of ${total_balance}.'])
+                    _put('credit', 0)
+		    _put('checking', get('checking') - responder.slots['total_balance'])
+	    #handle the system entity of the exact dollar amount they want to pay off
+            else:
+                responder.slots['amount'] = entity['value'][0]['value']
+                responder.reply(['Ok we have scheduled your credit card payment for {amount}'])
+                _put('credit', _get('credit') - entity['value'][0]['value'])
+		_put('checking', get('checking') - responder.slots['amount'])
+	responder.listen() 
+
+
+
+For a more realistic implementation of ``pay_creditcard`` that deals with varied user flows and the full code behind all the dialogue state handlers, see the ``__init__.py`` file in the blueprint folder.
+
+
+5. Slot Filling
```

Comment:
@abhijeetsidhu Could make this 4.1 or just an non-numbered sub-section under **4. Dialogue States**? We want to stick to 10 total sections since they have a one-to-one correspondence with 10-step guide in the documentation.

Reply:
Sure will make it a non-numbered sub-section.